src/app/task.service.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Task } from './task-full-descr-teacher/task-full-description.type';
+import { Observable } from 'rxjs';
+import { TaskFullDStudent } from './task-sending-page/task-sending-page.type';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TaskService {
+  constructor(private http: HttpClient) {
+  }
+
+  getTeacherFullDescription(id: string): Observable<Task> {
+    const headers = new HttpHeaders();
+    headers.append('Access-Control-Allow-Headers', 'Content-Type');
```

Comment:
I think it is better to do something like this: 
headers = new HttpHeaders();
constructor(private http: HttpClient) {
  this.headers.append('Access-Control-Allow-Headers', 'Content-Type');
  this.headers.append('Access-Control-Allow-Origin', 'http://localhost:4200');
}

Reply:
thanks for code